### PR TITLE
Add more tests.

### PR DESCRIPTION
**Pull request by bsansouci**
Here's a list of the added tests:
- parent kill child Spring
- TS unmount/mount correctly (and https://github.com/chenglou/react-motion/issues/77)
- init value + final value
- TS init value + final value

### CODE REVIEW

File: test/Spring-test.js [hunk 1]
```diff
@@ -391,4 +391,239 @@ describe('Spring', () => {
...
+        }
+        return this.state.isAlive && <App kill={() => this.setState({isAlive: false})}/>;
+      },
+    });
+
+    TestUtils.renderIntoDocument(<Parent />);
+
+    expect(count).toEqual([0]);
+    mockRaf.manySteps(4);
+    shouldKill = false;
```
Comment (chenglou) [inline test/Spring-test.js]: ?

Comment (bsansouci) [inline test/Spring-test.js]: This is just testing for that one edge case of rendering something that has been unmounted.
